src/minter-contract-utils/build.rs

```diff
@@ -88,3 +115,121 @@ fn get_solidity_contract<'a>(
 fn set_var(key: &str, value: &str) {
     println!("cargo:rustc-env={key}={value}");
 }
+
+struct SolidityFileTimesWatcher {
+    subpaths: Vec<PathBuf>,
+    times_file_path: PathBuf,
+    solidity_output_dir: PathBuf,
+}
+
+impl SolidityFileTimesWatcher {
```

Comment:
The logic could be better isolated this way:
- SolidityFileTimesWatcher to SolidityBuilder
- make save_file_times fn private
- add a new fn "build_updated_contracts" 
- in the `main` fn, simply call build_updated_contracts
- `save_file_times()` should not be called externally

And add tests! an error here could lead to the deployment of a wrong contract